src/stm32/Makefile

```diff
@@ -21,6 +21,8 @@ MCU_UPPER := $(shell echo $(CONFIG_MCU) | tr a-z A-Z | tr X x)
 CFLAGS-$(CONFIG_MACH_STM32F0) += -mcpu=cortex-m0 -Ilib/stm32f0/include
 CFLAGS-$(CONFIG_MACH_STM32F103) += -mcpu=cortex-m3
 CFLAGS-$(CONFIG_MACH_N32G45x) += -mcpu=cortex-m4 -Ilib/n32g45x/include
+CFLAGS-$(CONFIG_MACH_AT32F415) += -mcpu=cortex-m4
```

Comment:
This part is a bit confusing for me... the STM32F103xx have cortex-m3, whilst the AT32F41x have cortex-m4.
Are you sure using the STM32F103x header files is a good idea?
Perhaps it'd be safer to bring in the AT32F41x header files from the vendor themselves?
If the names within the header files are identical, then it shouldn't make any difference.  But is less likely to cause confusion in the future if there are some slight differences (i.e. some cortex-m3 register for TSC or such that is used, and suddenly breaks the AT32F controllers)